### DISCUSSION TITLE
Are DB triggers supported in PS/Vitess?

### DISCUSSION THREAD

**Discussion by siegerts** [planetscale/discussion#189] [Q&A] [answered]
Are DB triggers supported in PS/Vitess? If so, is this possible through client tools (`pscale` CLI, MySQL Workbench, etc)?  I can't seem to create any new triggers yet running `SHOW TRIGGERS;` is possible.

**Comment by orware** [comment, marked as answer]
Hello @siegerts,

This isn't something that we advertise as being a working option within PlanetScale due to the information shared here:
https://vitess.io/docs/13.0/reference/compatibility/mysql-compatibility/#stored-procedures
and
https://stackoverflow.com/questions/64064665/triggers-in-vitess-distributed-mysql

(Technically it's possible if you have direct access to the underlying MySQL instances and are running your own Vitess instances, but since that's not something we expose with our PlanetScale databases, it's not something we can recommend on our hosted cloud platform).

Instead, we would recommend that sort of logic that might normally go into a stored procedure (or trigger) to go into something within your application-level code (which also helps on the horizontally scaling side of things...keeping the PlanetScale side more focused on the data itself, and the application side managing the actions occurring for that data, so the two can more easily be scaled independently).

I hope this answer is helpful! (We are planning to add more details about this within the PlanetScale Docs as well in the near future to help make this more apparent up front for our potential users).

**Reply by siegerts** [reply]
Thanks for the reply! That makes sense. Just wanted to make sure that I wasn't missing something along the way.
